airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -22,32 +22,71 @@
 # SOFTWARE.
 #
 
+import json
+import time
 from abc import ABC, abstractmethod
 from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union
 
 import requests
 from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.exceptions import RequestBodyException
+from airbyte_cdk.sources.streams.http.http import BODY_REQUEST_METHODS
 from source_amazon_seller_partner.auth import AWSSignature
 
 REPORTS_API_VERSION = "2020-09-04"
 ORDERS_API_VERSION = "v0"
 
 
 class AmazonSPStream(HttpStream, ABC):
-    page_size = 100
     data_field = "payload"
 
-    def __init__(self, url_base: str, aws_signature: AWSSignature, replication_start_date: str, *args, **kwargs):
+    def __init__(
+        self, url_base: str, aws_signature: AWSSignature, replication_start_date: str, marketplace_ids: List[str], *args, **kwargs
+    ):
         super().__init__(*args, **kwargs)
 
         self._url_base = url_base
         self._aws_signature = aws_signature
         self._replication_start_date = replication_start_date
+        self.marketplace_ids = marketplace_ids
 
     @property
     def url_base(self) -> str:
         return self._url_base
 
+    def _create_prepared_request(
+        self, path: str, method: str = None, headers: Mapping = None, params: Mapping = None, json: Any = None, data: Any = None
```

Comment:
It's a little bit confusing to have parameter named `json` and to have `import json`.